clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/CopybookDownloader.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import { CopybookName } from "../CopybookDownloadService";
+
+export interface CopybookDownloader {
+  /**
+   * Downloads a file based on the passed parameters
+   *
+   * @param copybookName Copybook to be downloaded.
+   * @param documentUri cobol programs which needs copybook
+   * @param addressLocator locator of the remote copybook.
+   * @param callback callback function
+   */
+  downloadCopybook(
+    copybookName: CopybookName,
+    documentUri: string,
+    addressLocator: unknown,
+    callback?: () => void,
+  ): boolean | Promise<boolean>;
+
+  /**
+   * Checks if the file could be downloaded based on the passed parameters
+   * @param copybookName Copybook to be downloaded.
+   * @param documentUri cobol programs which needs copybook
+   * @param addressLocator locator of the remote copybook.
+   */
+  isEligibleForDownload(
```

Comment:
Is this part of the interface just to facilitate testing?

Reply:
No not actually, I thought it make sense to call a validate before download for each validator.